common/json/json.cpp

```diff
@@ -45,6 +48,56 @@ error::Error MakeError(JsonErrorCode code, const string &msg) {
 	return error::Error(error_condition(code, JsonErrorCategory), msg);
 }
 
+
+ExpectedStringVector ToStringVector(const json::Json &j) {
+	if (!j.IsArray()) {
+		return expected::unexpected(
+			MakeError(JsonErrorCode::ParseError, "The JSON object is not an array"));
+	}
+	vector<string> vector_elements {};
+	size_t vector_size {j.GetArraySize().value()};
+	for (size_t i = 0; i < vector_size; ++i) {
+		auto element = j.Get(i).and_then([](const json::Json &j) { return j.GetString(); });
+		if (!element) {
+			return expected::unexpected(MakeError(
+				JsonErrorCode::ParseError, "Unexpected element type in the string array"));
```

Comment:
These aren't really `ParseError`s though, are they? More like `WrongFormatError` or something like that.

Goes for all of the errors below.

Reply:
That makes it easy then! :tada: